### PR TITLE
Feat/pro 5078/ostree secondary

### PR DESCRIPTION
**Pull request by patriotyk**

### CODE REVIEW

File: src/aktualizr_secondary/aktualizr_secondary_config.cc [hunk 1]
```diff
@@ -48,6 +48,30 @@ void AktualizrSecondaryUptaneConfig::writeToStream(std::ostream& out_stream) con
   writeOption(out_stream, key_type, "key_type");
 }
 
+void PackageConfig::writeToStream(std::ostream& out_stream) const {
```
Comment (lbonn) [inline src/aktualizr_secondary/aktualizr_secondary_config.cc]: These two methods + `std::ostream& operator<<(std::ostream& os, PackageManager pm)` are already implemented in `config.cc` and `aktualizr-secondary` successfully links even when these implementations are removed.

Any specific reason to also define them here?

File: src/aktualizr_secondary/aktualizr_secondary.cc [hunk 2]
```diff
@@ -74,15 +75,43 @@ std::pair<KeyType, std::string> AktualizrSecondary::getPublicKeyResp() const {
...
+  }
+  if (target_found) {
+    if (target->format().empty() || target->format() == "OSTREE") {
+#ifdef BUILD_OSTREE
+      OstreeManager::pull(config_.pacman, keys_, target->sha256Hash());
+#else
+      LOG_ERROR << "Could not pull OSTree target. Aktualizr was built without OSTree support!";
+#endif
+    }
+    pacman->install(*target);
```
Comment (OYTIS) [inline src/aktualizr_secondary/aktualizr_secondary.cc]: Installation should only happen in `sendFirmwareResp`. It gets TLS credentials as a payload and without them OSTree will not be able to download the image.

Comment (patriotyk) [reply src/aktualizr_secondary/aktualizr_secondary.cc]: Ok

File: src/aktualizr_secondary/aktualizr_secondary_config.cc (hunk 1 above)
Comment (patriotyk) [reply src/aktualizr_secondary/aktualizr_secondary_config.cc]: Sorry, I just temporarry copied it.

File: src/aktualizr_secondary/aktualizr_secondary.cc [hunk 3]
```diff
@@ -74,16 +75,43 @@ std::pair<KeyType, std::string> AktualizrSecondary::getPublicKeyResp() const {
...
+        detected_attack_ = "Duplicate entry for this ECU";
+        break;
+      }
+      target_found = true;
+      target = boost::movelib::make_unique<Uptane::Target>(*it);
+    }
+  }
+  return true;
+}
+#ifdef BUILD_OSTREE
```
Comment (lbonn) [inline src/aktualizr_secondary/aktualizr_secondary.cc]: `sendFirmwareOstreeResp` instead of `sendFirmwareOstreResp` would appease me :).

**Comment by patriotyk** [comment]
I think the treehub url should be passed from primary during the provisoning

**Comment by cajun-rat** [comment]
Works for me

**Comment by cajun-rat** [comment]
We'll add the treehub URL in a second PR

**Pull request merged**